pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +624,282 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
 	if err != nil {
 		return err
 	}
-	return c.Do(request, 200, nil)
+	expectedStatus := []int{200}
+	return c.Do(request, expectedStatus, nil)
+}
+
+//MCServerRolesInfo is the struct for the server roles of the localhost, in this case SplunkMonitoringConsole
+type MCServerRolesInfo struct {
+	ServerRoles []string `json:"server_roles"`
+}
+
+//MCDistributedPeers is the struct for information about distributed peers of the monitoring console
+type MCDistributedPeers struct {
+	ClusterLabel []string `json:"cluster_label"`
+	ServerRoles  []string `json:"server_roles"`
+}
+
+//AutomateMCApplyChanges change the state of new indexers from "New" to "Configured" and add them in monitoring console asset table
+func (c *SplunkClient) AutomateMCApplyChanges(mock bool) error {
+	if mock {
+		return nil
+	}
+	var configuredPeers, indexerMemberList, licenseMasterMemberList string
+	apiResponseServerRoles, err := c.GetMonitoringconsoleServerRoles()
+	if err != nil {
+		return err
+	}
+
+	apiResponseMCDistributedPeers := struct {
```

Comment:
I guess these few lines should be a separate function similar to what you did for `GetMonitoringconsoleServerRoles`.